### DISCUSSION TITLE
I cannot git push into github refusing to allow a Personal Access Token to create or update workflow

### DISCUSSION THREAD

**Discussion by kristijorgji** [community/community#26254] [npm]
<p>Hi, after adding in my local repository .github and one workflow I cannot push anymore to github.</p>
<p>Doing git push I get this error</p>
<pre><code class="lang-auto">! [remote rejected] ok -&gt; ok (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/checks.yaml` without `workflow` scope)
error: failed to push some refs to
</code></pre>
<p>The content of .github/workflows/checks.yaml are</p>
<pre><code class="lang-auto">name: Checks
on:
  pull_request:
    types: [opened, synchronize]
jobs:
  check:
    runs-on: ubuntu-latest

    steps:
      - name: "Checkout" # Download code from the repository
        uses: actions/checkout@v2 # Public action
        with:
          fetch-depth: 0 # Checkout all branches and tags

      - name: "Use NodeJS" # Setup node using version 14
        uses: actions/setup-node@v2 # Public action
        with:
          node-version: '16.4.0'

      - name: "Setup npm" # Add our registry to npm config
        run: | # Custom action
          npm set @kristijorgji:registry=https://npm.pkg.github.com/kristijorgji
          npm set "//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_TOKEN }}"
          - name: Install dependencies

      run: | # Install and link dependencies
        yarn install --frozen-lockfile
        npx lerna bootstrap

      - name: Run tests # Run tests of all packages
        run: npx lerna exec npm run test

</code></pre>

**Comment by Simran-B** [comment, marked as answer]
<blockquote>
<p>refusing to allow a Personal Access Token to create or update workflow <code>.github/workflows/checks.yaml</code> without <code>workflow</code> scope)</p>
</blockquote>
<p>Sounds like the PAT misses the necessary permission? Make sure it has the workflow scope ticked.</p>

**Reply by linonetwo** [reply]
No, I'm using Github Desktop, where to modify its PAT?

**Reply by linonetwo** [reply]
Check your repo url, to see if there is a bad PAT already inside it.

**Comment by kristijorgji** [comment]
<p>Thank you yes I created a new personal token then was able to push and create pull request after modifying .github/workflow files<br>
(i had old token locally set up in mac keychain without workflow permissions before and therefore having the error)</p>

**Reply by LakshmiSanikommu** [reply]
hey , i have generated the personal token with workflow access , can i know where i need to add this to get accessible for all repos ?

**Reply by brainartfu** [reply]
Me too. I have generated the personal token with workflow access, But when I push to my repository, It keep same error.

**Reply by ArthurMro** [reply]
@LakshmiSanikommu @brainartfu I struggled with that part too. Here is the step by step, assuming you have already build your local repo and want to push it to a remote repo.
1) create a PAT (personal access token): [official doc here](https://docs.github.com/en/enterprise-server@3.4/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). Make sure to tick the box "workflow" when creating it.
2) In the terminal, instead of the classic 
`git remote add origin https://github.com/<account>/<repo>.git` 
swap it by 
`git remote add origin https://<PAT>@github.com/<account>/<repo>.git`
3) Follow-up with the classic `git branch -M main` and `git push -u origin main`

That worked for me. Hopefully for you too.

**Reply by Ambro19** [reply]
@cybr71 @kristijorgji There are two types of PAT (Fine-grained and Tokens(classic)). Please can you tell us which on to choose. I have generated PAT for both. But I am not sure which one to use.

**Reply by january-api** [reply]
> @cybr71 @kristijorgji There are two types of PAT (Fine-grained and Tokens(classic)). Please can you tell us which on to choose. I have generated PAT for both. But I am not sure which one to use.

For simplicity, use Token(classic).

**Reply by LWJcoder** [reply]
> @LakshmiSanikommu @brainartfu I struggled with that part too. Here is the step by step, assuming you have already build your local repo and want to push it to a remote repo.
> 
> 1. create a PAT (personal access token): [official doc here](https://docs.github.com/en/enterprise-server@3.4/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). Make sure to tick the box "workflow" when creating it.
> 2. In the terminal, instead of the classic
>    `git remote add origin https://github.com/<account>/<repo>.git`
>    swap it by
>    `git remote add origin https://<PAT>@github.com/<account>/<repo>.git`
> 3. Follow-up with the classic `git branch -M main` and `git push -u origin main`
> 
> That worked for me. Hopefully for you too.

The second step should be like this:git remote set-url origin https://your-username:your-new-token@github.com/your-repo.git

**Reply by brianMutea** [reply]
> > @LakshmiSanikommu @brainartfu I struggled with that part too. Here is the step by step, assuming you have already build your local repo and want to push it to a remote repo.
> > 
> > 1. create a PAT (personal access token): [official doc here](https://docs.github.com/en/enterprise-server@3.4/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). Make sure to tick the box "workflow" when creating it.
> > 2. In the terminal, instead of the classic
> >    `git remote add origin https://github.com/<account>/<repo>.git`
> >    swap it by
> >    `git remote add origin https://<PAT>@github.com/<account>/<repo>.git`
> > 3. Follow-up with the classic `git branch -M main` and `git push -u origin main`
> > 
> > That worked for me. Hopefully for you too.
> 
> The second step should be like this:git remote set-url origin https://your-username:your-new-token@github.com/your-repo.git

This second step format you have given at the end seems not to work. Maybe wrongly formatted: Use this: `git remote add origin https://<PAT>@github.com/<your-username>/<repo>.git `

**Comment by cybr71** [comment]
<p>Hello kristijorgji,<br>
Greetings!</p>
<p>You have to generate new token with workflow scope permission. It’ll resolve your issue.<br>
Please make sure workflow is selected while creating new token.</p>
<p>Thanks!<br>
Best Regards,<br>
Md. Saidur Rahman</p>

**Reply by piotr-cz** [reply]
I was able to update existing token by adding the `workflow` rule

**Reply by AhmadMujtaba200210** [reply]
yeah but how to use it?

**Comment by invzbl3** [comment]
I've reproduced the same issue and if someone needs to refresh steps how to create a personal access token in GitHub, [this video](https://youtu.be/SzrETQdGzBM?t=49) will be helpful for you.

**Comment by imukherjee-sndk** [comment]
Easiest fix seems like just updating the existing token with additional scope 'workflow' ... worked for me.

**Comment by 01walid** [comment]
For those using `gh` CLI and authenticated against Github with `gh auth login` flow, you'll need to pass `-s` to request additional scopes. 

So re-authenticate again with `gh auth login -s workflow`.

**Reply by Zaulao** [reply]
This should be the highlighted answer! Thanks!

**Comment by chylex** [comment]
I already had `workflow` enabled in my token, but it didn't work... I had to configure Windows Credential Manager to use `PersonalAccessToken` as the username and the token as the password.

I don't get why this is a problem when I have GitHub Actions disabled on all my repositories. I'm trying to update a fork, the fork added a workflow file, and suddenly I can't push into my own fork to make a PR against the most recent version of the upstream branch. Pretty annoying...

**Comment by schakrabortygithub** [comment]
Recently I faced this issue.
Simplest solution is to go the Personal Access Token page of your GitHub profile and modify that token to include the 'workflow' scope.
Go to Settings-> Developer setttings->Personal access tokens-> Tokens(classic)

![Screen Shot 2024-01-26 at 1 51 01 AM](https://github.com/community/community/assets/35871741/40d95f97-85bb-42bf-90bf-96df8325803b)

**Reply by FaizIkhwan** [reply]
Thank you, this solve my issue.

**Reply by EliLazarus** [reply]
I would mark this one, super user friendly to go through the Github website.
